graph/resolver/participant.resolvers.go

```diff
@@ -127,7 +150,42 @@ func (r *participantResolver) UserProfile(ctx context.Context, obj *model.Partic
 	return userProfile, nil
 }
 
+func (r *participantResolver) AnonDisplayName(ctx context.Context, obj *model.Participant) (*string, error) {
+	if !obj.IsAnonymous {
+		return nil, nil
+	}
+
+	if obj.Discussion == nil && obj.DiscussionID != nil {
+		disc, err := r.resolveDiscussionByID(ctx, *obj.DiscussionID)
+		if err != nil {
+			return nil, err
+		}
+		obj.Discussion = disc
+	}
+
+	if obj.Discussion == nil {
+		return nil, fmt.Errorf("Could not find associated discussion")
+	}
+
+	hashAsInt64 := generateParticipantSeed(*obj.Discussion, *obj)
+	fullDisplayName := util.GenerateFullDisplayName(hashAsInt64)
+	return &fullDisplayName, nil
+}
+
 // Participant returns generated.ParticipantResolver implementation.
 func (r *Resolver) Participant() generated.ParticipantResolver { return &participantResolver{r} }
 
 type participantResolver struct{ *Resolver }
+
+// !!! WARNING !!!
+// The code below was going to be deleted when updating resolvers. It has been copied here so you have
+// one last chance to move it out of harms way if you want. There are two reasons this happens:
+//  - When renaming or deleting a resolver the old code will be put in here. You can safely delete
+//    it when you're done.
+//  - You have helper methods in this file. Move them out to keep these resolver files clean.
+func generateParticipantSeed(discussion model.Discussion, participant model.Participant) uint64 {
```

Comment:
this will be removed but is called in a couple places above. can we move into the backend or util packages?